continuous_integration/run_tests.sh

```diff
@@ -19,15 +19,10 @@ python -c "import multiprocessing as mp; print('multiprocessing.cpu_count():', m
 python -c "import joblib; print('joblib.cpu_count():', joblib.cpu_count())"
 
 if [[ "$SKLEARN_TESTS" != "true" ]]; then
-    if [ "$COVERAGE" == "true" ]; then
-        # Enable coverage-related options. --cov-append is needed to combine
-        # the test run and the test-doc run coverage.
-        export PYTEST_ADDOPTS="--cov=joblib --cov-append"
-    fi
+    pytest joblib -vl --timeout=120 --cov=joblib --cov-report xml
 
-    pytest joblib -vl --timeout=120 --junitxml="${JUNIT_XML}"
     # doctests are not compatile with default_backend=threading
-    if [ "$JOBLIB_TESTS_DEFAULT_PARALLEL_BACKEND" != "threading"]; then
+    if [[ "$JOBLIB_TESTS_DEFAULT_PARALLEL_BACKEND" != "threading" ]]; then
```

Comment:
Looks like because the `]` is missing a space before we were in some kind of a weird bash situation and we were never running the doctests from doc rst files? No idea ... I don't want to know to be honest. This is why I had to change `doc/memory.rst` below.